custom_components/ha_washdata/ml/matching_tuner.py

```
@@ -0,0 +1,180 @@
+        )
+    return by_profile
+
+
+def _snaps(by_profile: dict[str, list[dict]], exclude: tuple[str, int] | None) -> list[dict[str, Any]]:
+    snaps = []
+    for name, items in by_profile.items():
+        curves, durs = [], []
+        for idx, it in enumerate(items):
+            if exclude is not None and (name, idx) == exclude:
+                continue
+            curves.append(it["rs"]); durs.append(it["dur"])
```

Ruff (0.15.20)

[error] Multiple statements on one line (semicolon)

(E702)